### PR TITLE
Update validation dependencies of meta-plugins and C-API

### PR DESCRIPTION
**Pull request by peterchen-intel**
### Details:
Updated per following realities:
- meta-plugins are tested with core, TEMPLATE plugin and mock, are not tested with hardware plugins in pre-commits
- meta-plugins are using internal properties from CPU/GPU plugins (inside inference/)
- GPU enables AUTO BATCH by default for throughput mode
- AUTO is default device
- C-API tests remote context if GPU plugin exists

### Tickets:
 - *CVS-129319,CVS-129332,CVS-129316,CVS-129317*

### CODE REVIEW

File: .github/components.yml [hunk 1]
```diff
@@ -34,41 +34,34 @@ CPU:
...
   build:
-    - HETERO
-    - AUTO_BATCH
+    - AUTO
     - TEMPLATE
     - IR_FE
 
 GPU:
   build:
-    - HETERO
```
Comment (riverlijunjie) [inline .github/components.yml:44]: Why remove HETERO?

Comment (peterchen-intel) [reply .github/components.yml:44]: En, if HETERO is using GPU internal header files. We should enable the build.

Comment (ilya-lavrenov) [reply .github/components.yml:44]: GPU headers are part of `inference` component:
https://github.com/openvinotoolkit/openvino/blob/637920ce9154f47ca2c1992753406d8dbca39649/.github/labeler.yml#L85-L87
which triggers rebuild and revalidate of other dependent components:
https://github.com/openvinotoolkit/openvino/blob/637920ce9154f47ca2c1992753406d8dbca39649/.github/components.yml#L23-L24

**Comment by ilya-lavrenov** [comment]
is it still valid?
please, take into account that we can to avoid extra dependencies between components.. and that is why we asked to remove meta-plugins testing from HW plugin test suites.

**Comment by peterchen-intel** [comment]
> is it still valid? please, take into account that we can to avoid extra dependencies between components.. and that is why we asked to remove meta-plugins testing from HW plugin test suites.

In this PR,  we have remove AUTO/MULTI/BATCH/HTERO validation from any changes of HW plugins and TEMPLATE plugin. Please continue the review and merge if there is not other concerns.

File: .github/components.yml (hunk 1 above)
Comment (peterchen-intel) [reply .github/components.yml:44]: @ilya-lavrenov I have added AUTO/BATCH/HETERO back to the build of CPU/GPU changes, the reason is to keep same build configuration as release. They don't add too much time especially if we are using cache.

File: .github/components.yml [hunk 2]
```diff
@@ -34,41 +34,37 @@ CPU:
...
   revalidate:
-    - CPU
-    - GPU
-    - HETERO
     - AUTO_BATCH
-    - TEMPLATE
     - C_API
     - Python_API
   build:
+    - TEMPLATE
```
Comment (peterchen-intel) [inline .github/components.yml:67]: - HETERO

Comment (peterchen-intel) [inline .github/components.yml:67]:
```suggestion
    - TEMPLATE
    - HETERO
```

**Comment by peterchen-intel** [comment]
CAPI is in https://github.com/openvinotoolkit/openvino/pull/23414

**Pull request closed without merging**